### PR TITLE
Refactor `FormulaCreator` args and call `parse_url` automatically

### PR DESCRIPTION
**Pull request by abitrolly**
- [x] Have you followed the guidelines in our [Contributing](https://github.com/Homebrew/brew/blob/HEAD/CONTRIBUTING.md) document?
- [x] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/Homebrew/brew/pulls) for the same change?
- [x] Have you added an explanation of what your changes do and why you'd like us to include them?
- [x] Have you written new tests for your changes? [Here's an example](https://github.com/Homebrew/brew/blob/HEAD/Library/Homebrew/test/PATH_spec.rb).
- [x] Have you successfully run `brew style` with your changes locally?
- [x] Have you successfully run `brew typecheck` with your changes locally?
- [x] Have you successfully run `brew tests` with your changes locally?

-----

This PR makes all `FormulaCreator` params, except URL optional, which will make its call in tests less verbose. The changeset also calls `parse_url` in constructor to ensure that supplied URL is automatically parsed, which should happen anyway before formula is written (as discussed in https://github.com/Homebrew/brew/pull/20025/files#r2123010882).

### CODE REVIEW

File: Library/Homebrew/formula_creator.rb [hunk 1]
```diff
@@ -10,10 +10,10 @@ class FormulaCreator
     attr_accessor :name
 
     sig {
-      params(name: T.nilable(String), version: T.nilable(String), tap: T.nilable(String), url: String,
+      params(url: String, name: T.nilable(String), version: T.nilable(String), tap: T.nilable(String),
              mode: T.nilable(Symbol), license: T.nilable(String), fetch: T::Boolean, head: T::Boolean).void
     }
-    def initialize(name, version, tap:, url:, mode:, license:, fetch:, head:)
+    def initialize(url:, name: nil, version: nil, tap: nil, mode: nil, license: nil, fetch: false, head: false)
```
Comment (MikeMcQuaid) [inline Library/Homebrew/formula_creator.rb:23]: Don't set default arguments here unless they are needed by at least one caller.

File: Library/Homebrew/test/spec_helper.rb [hunk 2]
```diff
@@ -252,7 +252,7 @@
 
     # Link original API cache files to test cache directory.
     Pathname("#{ENV.fetch("HOMEBREW_CACHE")}/api").glob("*.json").each do |path|
-      FileUtils.ln path, HOMEBREW_CACHE/"api/#{path.basename}"
+      FileUtils.ln_s path, HOMEBREW_CACHE/"api/#{path.basename}"
```
Comment (MikeMcQuaid) [inline Library/Homebrew/test/spec_helper.rb]: This should be removed from here.

**Review by MikeMcQuaid — COMMENTED** [review]
Thanks for splitting this out. A few nits but looks good so far.

File: Library/Homebrew/formula_creator.rb (hunk 1 above)
Comment (abitrolly) [reply Library/Homebrew/formula_creator.rb:23]: @MikeMcQuaid keyword arguments without defaults will become required, which means tests will have to specify all these as `nil` anyway.

File: Library/Homebrew/test/spec_helper.rb (hunk 2 above)
Comment (abitrolly) [reply Library/Homebrew/test/spec_helper.rb]: Fixed.

File: Library/Homebrew/formula_creator.rb (hunk 1 above)
Comment (MikeMcQuaid) [reply Library/Homebrew/formula_creator.rb:23]: @abitrolly Feels like it might be nicer to just keep the original API then and not use keyword arguments?

Comment (abitrolly) [reply Library/Homebrew/formula_creator.rb:23]: @MikeMcQuaid the original API still uses required keyword arguments, except for `name` and `version`, but the only required parameter for `FormulaCreator` logic is URL, so the new API reflects that.

Comment (MikeMcQuaid) [reply Library/Homebrew/formula_creator.rb:23]: @abitrolly Sorry, I don't understand why this API is any better than what's there before. You'll need to try a bit harder to explain and convince me here.

Comment (abitrolly) [reply Library/Homebrew/formula_creator.rb:23]: If default are removed back to original API, then in tests, instead of this:
```
fc = described_class.new(url: test[:url])
```
It will have to be this
```
        fc = described_class.new(
          nil,
          nil,
          tap:     nil,
          url:     test[:url],
          name:    nil,
          version: nil,
          tap:     nil,
          mode: nil,
          license: nil,
          fetch:   false,
          head:    false,
        )```

Comment (MikeMcQuaid) [reply Library/Homebrew/formula_creator.rb:23]: Why can't they just keep using `name_from_url`? That method only exists for tests anyway, it seems. Why aren't you deleting it here if it's not used any more?

I'm getting pretty frustrated by the lack of attention to detail in these PRs @abitrolly. I'm not sure a PR like this adds any value to users and it feels like I'm putting in the effort verifying things that you should be doing. This is not your first PR, you can do better than this by now.

Comment (abitrolly) [reply Library/Homebrew/formula_creator.rb:23]: @MikeMcQuaid the `name_from_url` exists to simplify the `parse_url` logic, which also tries to detect version from the URL.

I tried to separate tests from https://github.com/Homebrew/brew/pull/20025 to ensure they pass after the PR. Let me make it more complete by including version in test data as well.

Comment (abitrolly) [reply Library/Homebrew/formula_creator.rb:23]: @MikeMcQuaid I've committed the complete test for the `FormulaCreator.new` that checks correctness of both versions and names parsed from URL. The structure allows to set only needed input params and check only relevant properties. Please take a look.

Comment (MikeMcQuaid) [reply Library/Homebrew/formula_creator.rb:23]: @abitrolly I can see a complete test and other uses of `name_from_url` now, thanks. Given that, I think it makes a lot more sense for `name_from_url` to get folded into `parse_url`. It may also make sense to fold in `parse_url` into the constructor.

In general: methods that are only called once make sense to be combined unless:
- the combined method is very very very long (and RuboCop will tell you off for this)
- the individual methods are called separately

I don't think that is the case for either here.

File: Library/Homebrew/test/formula_creator_spec.rb [hunk 3]
```diff
@@ -3,28 +3,61 @@
...
+      },
+    }
 
-  it "gets name from generic tarball URL" do
-    t = described_class.name_from_url("http://digit-labs.org/files/tools/synscan/releases/synscan-5.02.tar.gz")
-    expect(t).to eq("synscan")
+    tests.each do |description, test|
+      it "parses #{description}" do
+        params = test[:params]
+        fc = described_class.new(**params)
```
Comment (MikeMcQuaid) [inline Library/Homebrew/test/formula_creator_spec.rb]:
```suggestion
        fc = described_class.new(url: test[:params][:url])
```
as you never pass any other parameters. would also probably be nice to flatten out `params` given that.

File: Library/Homebrew/test/formula_creator_spec.rb [hunk 4]
```diff
@@ -3,28 +3,61 @@
...
+        test[:expected].each do |key, value|
+          expect(fc.instance_variable_get("@#{key}")).to eq(value)
```
Comment (MikeMcQuaid) [inline Library/Homebrew/test/formula_creator_spec.rb]: Don't use `fc.instance_variable_get` like this in tests. If you need to access these variables, provide an `attr_reader` for `version`. There's already one for `name`.

Comment (abitrolly) [reply Library/Homebrew/test/formula_creator_spec.rb]: Done. A little bit more verbose and a little less cryptic.

File: Library/Homebrew/formula_creator.rb (hunk 1 above)
Comment (abitrolly) [reply Library/Homebrew/formula_creator.rb:23]: Makes sense. Folded `name_from_url` into `parse_url`. I would leave folding everything into the constructor for another PR. For now calling `parse_url` makes it explicit that no other logic is happening, except detecting missing values from URL.

Comment (abitrolly) [reply Library/Homebrew/formula_creator.rb:23]: @MikeMcQuaid thanks for the review. I think it is ready now.

File: Library/Homebrew/test/formula_creator_spec.rb [hunk 5]
```diff
@@ -3,28 +3,61 @@
...
-  it "gets name from GitHub download URL" do
-    t = described_class.name_from_url("https://github.com/stella-emu/stella/releases/download/6.7/stella-6.7-src.tar.xz")
-    expect(t).to eq("stella")
-  end
+  describe ".new" do
+    tests = {
+      "generic tarball URL": {
+        params:   {
+          url: "http://digit-labs.org/files/tools/synscan/releases/synscan-5.02.tar.gz",
+        },
```
Comment (MikeMcQuaid) [inline Library/Homebrew/test/formula_creator_spec.rb]:
```suggestion
        url: "http://digit-labs.org/files/tools/synscan/releases/synscan-5.02.tar.gz",
```
and same below

File: Library/Homebrew/test/formula_creator_spec.rb [hunk 6]
```diff
@@ -3,28 +3,61 @@
...
+      },
+    }
 
-  it "gets name from generic tarball URL" do
-    t = described_class.name_from_url("http://digit-labs.org/files/tools/synscan/releases/synscan-5.02.tar.gz")
-    expect(t).to eq("synscan")
+    tests.each do |description, test|
+      it "parses #{description}" do
+        fc = described_class.new(url: test[:params][:url])
```
Comment (MikeMcQuaid) [inline Library/Homebrew/test/formula_creator_spec.rb]:
```suggestion
        fc = described_class.new(url: test.fetch(url))
```

File: Library/Homebrew/test/formula_creator_spec.rb [hunk 7]
```diff
@@ -3,28 +3,61 @@
...
+        test[:expected].each do |key, value|
+          expect(fc.name).to eq(value) if key == :name
+          expect(fc.version).to eq(value) if key == :version
+        end
```
Comment (MikeMcQuaid) [inline Library/Homebrew/test/formula_creator_spec.rb]: Why was this a loop?
```suggestion
        result = { name: fc.name, version: fc.version }
        expect(result).to eq(test.fetch(:expected))
```

File: Library/Homebrew/formula_creator.rb [hunk 8]
```diff
@@ -8,12 +8,13 @@ module Homebrew
...
       @name = name
       @version = Version.new(version) if version
```
Comment (MikeMcQuaid) [inline Library/Homebrew/formula_creator.rb]:
```suggestion
      @name = name.presence
      @version = Version.new(version) if version.present?
```

File: Library/Homebrew/formula_creator.rb [hunk 9]
```diff
@@ -8,12 +8,13 @@ module Homebrew
...
       @tap = Tap.fetch(tap || "homebrew/core")
```
Comment (MikeMcQuaid) [inline Library/Homebrew/formula_creator.rb]: Move `@url` below up so the variables match the parameters order. Use `.presence` on anything else that's a `T.nilable(String)` e.g. `mode`, `license`

Comment (MikeMcQuaid) [inline Library/Homebrew/formula_creator.rb]:
```suggestion
      @tap = Tap.fetch(tap.presence || "homebrew/core")
```

File: Library/Homebrew/formula_creator.rb [hunk 10]
```diff
@@ -8,12 +8,13 @@ module Homebrew
   # Class for generating a formula from a template.
```
Comment (MikeMcQuaid) [inline Library/Homebrew/formula_creator.rb:8]: Consider making this `typed: strict` which will handle the types better

File: Library/Homebrew/formula_creator.rb [hunk 11]
```diff
@@ -22,34 +23,34 @@ def initialize(name, version, tap:, url:, mode:, license:, fetch:, head:)
...
+          Regexp.last_match(1)
+        else
+          # e.g. http://digit-labs.org/files/tools/synscan/releases/synscan-5.02.tar.gz
+          pathver = Version.parse(stem).to_s
+          stem.sub(/[-_.]?#{Regexp.escape(pathver)}$/, "")
+        end
+        odebug "name from url: #{@name}"
+      end
+
       @version = Version.detect(@url) if @version.nil?
```
Comment (MikeMcQuaid) [inline Library/Homebrew/formula_creator.rb]:
```suggestion
      @version ||= Version.detect(@url)
```

File: Library/Homebrew/formula_creator.rb [hunk 12]
```diff
@@ -22,34 +23,34 @@ def initialize(name, version, tap:, url:, mode:, license:, fetch:, head:)
...
+        elsif @url =~ %r{github\.com/\S+/(\S+)/(archive|releases)/}
+          # e.g. https://github.com/stella-emu/stella/releases/download/6.7/stella-6.7-src.tar.xz
+          Regexp.last_match(1)
+        else
+          # e.g. http://digit-labs.org/files/tools/synscan/releases/synscan-5.02.tar.gz
+          pathver = Version.parse(stem).to_s
+          stem.sub(/[-_.]?#{Regexp.escape(pathver)}$/, "")
+        end
+        odebug "name from url: #{@name}"
+      end
```
Comment (MikeMcQuaid) [inline Library/Homebrew/formula_creator.rb]: This will work once you do `name.presence` above

```suggestion
      @name ||= begin
        stem = Pathname.new(@url).stem
        name = if stem.start_with? "index.cgi"
          # special cases first
          # gitweb URLs e.g. http://www.codesrc.com/gitweb/index.cgi?p=libzipper.git;a=summary
          stem.rpartition("=").last
        elsif @url =~ %r{github\.com/\S+/(\S+)/(archive|releases)/}
          # e.g. https://github.com/stella-emu/stella/releases/download/6.7/stella-6.7-src.tar.xz
          Regexp.last_match(1)
        else
          # e.g. http://digit-labs.org/files/tools/synscan/releases/synscan-5.02.tar.gz
          pathver = Version.parse(stem).to_s
          stem.sub(/[-_.]?#{Regexp.escape(pathver)}$/, "")
        end
        odebug "name from url: #{name}"
        name
      end
```

File: Library/Homebrew/test/formula_creator_spec.rb (hunk 7 above)
Comment (abitrolly) [reply Library/Homebrew/test/formula_creator_spec.rb]: It is loop for testing each field, because some tests don't contain version field.
```
       expected: {name: "lapce"}
            got: {name: "lapce", version: #<Version::NULL>}
```

File: Library/Homebrew/test/formula_creator_spec.rb (hunk 5 above)
Comment (abitrolly) [reply Library/Homebrew/test/formula_creator_spec.rb]: Done.

File: Library/Homebrew/test/formula_creator_spec.rb (hunk 6 above)
Comment (abitrolly) [reply Library/Homebrew/test/formula_creator_spec.rb]: Fixed.

File: Library/Homebrew/formula_creator.rb (hunk 9 above)
Comment (abitrolly) [reply Library/Homebrew/formula_creator.rb]: Done.

File: Library/Homebrew/formula_creator.rb (hunk 12 above)
Comment (abitrolly) [reply Library/Homebrew/formula_creator.rb]: Committed.

File: Library/Homebrew/test/formula_creator_spec.rb (hunk 7 above)
Comment (abitrolly) [reply Library/Homebrew/test/formula_creator_spec.rb]: @MikeMcQuaid without loop every test will need to explicitly specify defaults for values that are not tested, so I left the loop for now.. I've also added test for setting `head` for GitHub source URLs, and I don't see how to handle it better..

File: Library/Homebrew/formula_creator.rb (hunk 10 above)
Comment (abitrolly) [reply Library/Homebrew/formula_creator.rb:8]: Added some types. This reduced `typed: strict` from 32 errors down to 8. But without strict there is 1 problem now.

```ruby
formula_creator.rb:61: This code is unreachable https://srb.help/7006
    61 |      @version ||= Version.detect(@url)
                           ^^^^^^^^^^^^^^^^^^^^
    formula_creator.rb:61: This condition was always truthy (Version)
    61 |      @version ||= Version.detect(@url)
              ^^^^^^^^
  Got Version originating from:
    formula_creator.rb:26:
    26 |      @version = T.let(Version.new(version), Version) if version.present?
              ^^^^^^^^
    formula_creator.rb:61: This condition was always truthy (Version)
    61 |      @version ||= Version.detect(@url)
              ^^^^^^^^
  Got Version originating from:
    formula_creator.rb:26:
    26 |      @version = T.let(Version.new(version), Version) if version.present?
              ^^^^^^^^
Errors: 1
```

Comment (abitrolly) [reply Library/Homebrew/formula_creator.rb:8]: This typecheck for version is wrong - the code is reachable. Not sure how to fix it. @MikeMcQuaid any ideas?

File: Library/Homebrew/test/formula_creator_spec.rb [hunk 13]
```diff
@@ -3,28 +3,53 @@
...
+      },
+    }
 
-  it "gets name from generic tarball URL" do
-    t = described_class.name_from_url("http://digit-labs.org/files/tools/synscan/releases/synscan-5.02.tar.gz")
-    expect(t).to eq("synscan")
+    tests.each do |description, test|
+      it "parses #{description}" do
+        fc = described_class.new(url: test.fetch(:url))
+        test.fetch(:expected).each do |key, value|
```
Comment (MikeMcQuaid) [inline Library/Homebrew/test/formula_creator_spec.rb]: Stop looping here. There's absolutely no need. I've already said this. Please read what I've said.

You should verify the `version` and `head` every time. If they aren't specificed they should be verified to be falsey. This does not require a loop.

Comment (abitrolly) [reply Library/Homebrew/test/formula_creator_spec.rb]: @MikeMcQuaid fixed.

**Comment by abitrolly** [comment]
@MikeMcQuaid addressed all comments. What to do with `brew typecheck` failure? Should I revert types?

EDIT: Nevermind. This is ready for review.

File: Library/Homebrew/formula_creator.rb (hunk 10 above)
Comment (abitrolly) [reply Library/Homebrew/formula_creator.rb:8]: Fixed the type check.

```diff
-      @version = T.let(Version.new(version), Version) if version.present?
+      @version = T.let(Version.new(version), T.nilable(Version)) if version.present?
```

I don't get how Sorbet works - it looks like type won't be assigned if `version` is `nil`, but the typecheck passes.

**Comment by MikeMcQuaid** [comment]
@abitrolly Please stop pushing to this PR. I'm going to fix it.

**Comment by MikeMcQuaid** [comment]
@abitrolly Please test the latest version here. I've completed the typechecking and refactoring.

File: Library/Homebrew/dev-cmd/create.rb [hunk 14]
```diff
@@ -183,61 +183,62 @@ def create_formula
...
         )
-        fc.parse_url
+
         # ask for confirmation if name wasn't passed explicitly
         if args.set_name.blank?
-          print "Formula name [#{fc.name}]: "
-          fc.name = __gets || fc.name
+          print "Formula name [#{formula_creator.name}]: "
+          confirmed_name = __gets.presence
+          formula_creator.name = confirmed_name if confirmed_name.present?
```
Comment (abitrolly) [inline Library/Homebrew/dev-cmd/create.rb]:
```suggestion
          confirmed_name = __gets
          formula_creator.name = confirmed_name if confirmed_name.present?
```

File: Library/Homebrew/test/formula_creator_spec.rb [hunk 15]
```diff
@@ -6,48 +6,41 @@
...
     tests = {
       "generic tarball URL": {
-        url:      "http://digit-labs.org/files/tools/synscan/releases/synscan-5.02.tar.gz",
-        expected: {
-          name:    "synscan",
-          version: "5.02",
-        },
+        url:     "http://digit-labs.org/files/tools/synscan/releases/synscan-5.02.tar.gz",
+        name:    "synscan",
+        version: "5.02",
```
Comment (abitrolly) [inline Library/Homebrew/test/formula_creator_spec.rb:11]: Now it it not clear what is the fixture and what is the expectation. `url` can change during parsing.

File: Library/Homebrew/test/formula_creator_spec.rb [hunk 16]
```diff
@@ -6,48 +6,41 @@
...
-        expect(fc.version).to eq(ex[:version]) if ex.key?(:version)
-        expect(fc.head).to eq(ex[:head]) if ex.key?(:head)
+        formula_creator = described_class.new(url: test.fetch(:url))
+        expect(formula_creator.name).to eq(test.fetch(:name))
+        if (version = test[:version])
+          expect(formula_creator.version).to eq(version)
+        else
+          expect(formula_creator.version).to be_null
+        end
+        expect(formula_creator.head).to eq(test.fetch(:head, false))
```
Comment (abitrolly) [inline Library/Homebrew/test/formula_creator_spec.rb:43]: The original idea is fully declarative table test fixture. Dynamic enrichment of static test data with "default values" makes it less readable for me. If there is an error that sets `head` to nil, and test expects `false`, the test won't catch it.

The same with version here - if we need a test that checks version is null after parsing, it should be explicitly set in fixture. One test for URL without version is enough to check the behavior. No need to complicate the logic to repeat it for all URLs where the version is irrelevant. What could be relevant is to add a fixture entry that tests that version from params overrides the one parsed from URL.

File: Library/Homebrew/formula_creator.rb [hunk 17]
```diff
@@ -22,55 +22,66 @@ class FormulaCreator
...
+        github = GitHub.repository(user, repository) if fetch
       when %r{github\.com/(\S+)/(\S+)/(archive|releases)/}
         user = Regexp.last_match(1)
-        repo = Regexp.last_match(2)
-        @github = GitHub.repository(user, repo) if @fetch
+        repository = Regexp.last_match(2)
+        github = GitHub.repository(user, repository) if fetch
       end
+      @head = head
+      @github = T.let(github, T.untyped)
```
Comment (abitrolly) [inline Library/Homebrew/formula_creator.rb:74]: This looks like JSON like object from REST API.

**Review by abitrolly — COMMENTED** [review]
Now I see what you mean by combining parsing logic into constructor. The assignment of instance variables became more clear. Looks good.

I kind of disagree with the way test data is processed, but that's not a blocker. Just a personal preference from Go/Python side.

File: Library/Homebrew/dev-cmd/create.rb (hunk 14 above)
Comment (MikeMcQuaid) [reply Library/Homebrew/dev-cmd/create.rb]: @abitrolly Why does it make sense to accept a blank name from the user here?

File: Library/Homebrew/formula_creator.rb (hunk 17 above)
Comment (MikeMcQuaid) [reply Library/Homebrew/formula_creator.rb:74]: @abitrolly the API does not provide any guarantees so `T.untyped` must be used.

File: Library/Homebrew/test/formula_creator_spec.rb (hunk 15 above)
Comment (MikeMcQuaid) [reply Library/Homebrew/test/formula_creator_spec.rb:11]: What do you propose? `expected_name` and `expected_version`? Criticising without offering an alternative is not helpful.

File: Library/Homebrew/test/formula_creator_spec.rb (hunk 16 above)
Comment (MikeMcQuaid) [reply Library/Homebrew/test/formula_creator_spec.rb:43]: > The original idea is fully declarative table test fixture.

This is not something we do in Homebrew.

> Dynamic enrichment of static test data with "default values" makes it less readable for me.

As someone who is neither an expert in Ruby nor Homebrew, readability for you is not the desired outcome here.

> If there is an error that sets `head` to nil, and test expects `false`, the test won't catch it.

The Sorbet type system will catch it.

> The same with version here - if we need a test that checks version is null after parsing, it should be explicitly set in fixture.

I disagree.

> No need to complicate the logic to repeat it for all URLs where the version is irrelevant.

The version is never irrelevant.

> What could be relevant is to add a fixture entry that tests that version from params overrides the one parsed from URL.

Will review a follow-up PR to do that.

**Review by MikeMcQuaid — COMMENTED** [review]
@abitrolly Can you clarify whether you tested this or not? Please can we move this to a conclusion? For refactoring with no functionality change this PR has wasted a _lot_ of time and I am losing my patience with the amount of back and forth required here.

File: Library/Homebrew/dev-cmd/create.rb (hunk 14 above)
Comment (abitrolly) [reply Library/Homebrew/dev-cmd/create.rb]: @MikeMcQuaid according to [this blog](https://blog.appsignal.com/2018/09/11/differences-between-nil-empty-blank-and-present.html#present), `#present?` is `false` on `nil` and empty strings.

```
"".present?
=> false
 
nil.present?
=> false
```

So no need to check that `confirmed_name` is blank twice.

Comment (MikeMcQuaid) [reply Library/Homebrew/dev-cmd/create.rb]: Ok, I understand now, thanks.

File: Library/Homebrew/test/formula_creator_spec.rb (hunk 15 above)
Comment (abitrolly) [reply Library/Homebrew/test/formula_creator_spec.rb:11]: @MikeMcQuaid same structure as before https://github.com/Homebrew/brew/pull/20116/commits/85d48da3640fe7de496fd941375088c9baff3645#diff-e680f7a2b450b630d317f4b6f134aea581047c6bb36117b72b249d9da14869c8 

```ruby
    tests = {
      "generic tarball URL": {
        url:      "http://digit-labs.org/files/tools/synscan/releases/synscan-5.02.tar.gz",
        expected: {
          name:    "synscan",
          version: "5.02",
        },
      },
```

So each entry is:
```
testname: {
  url: xxx,
  expected: {
     name: xxx
     version: xxx
  }
}
```

Each test entry can be even more explicit about params and expectations.
```
testname: {
  params: {
    url: string,
  },
  expected: {
     name: xxx,
     version: xxx,
  }
}
```

Then the test data for version override will look like this.

```ruby
      "with version override": {
        params: {
           url:     "http://digit-labs.org/files/tools/synscan/releases/synscan-5.02.tar.gz",
           version: "3.40",
        },
        expected: {
           name:    "synscan",
           version: "3.40",
        }
      },
```

File: Library/Homebrew/test/formula_creator_spec.rb (hunk 16 above)
Comment (abitrolly) [reply Library/Homebrew/test/formula_creator_spec.rb:43]: Okay. Resolving.

**Comment by abitrolly** [comment]
@MikeMcQuaid I've tested it with `brew tests` and manually run with and without new name. Works as expected.

```
brew create https://github.com/buildpacks/pack/archive/refs/tags/v0.37.0.tar.gz -d
```

**Comment by abitrolly** [comment]
The conclusion is that it is ready long ago. )

**Comment by MikeMcQuaid** [comment]
Thanks @abitrolly.
